lib/cog/commands/pluck.ex

```diff
@@ -0,0 +1,115 @@
+defmodule Cog.Commands.Pluck do
+  use Spanner.GenCommand.Base, bundle: Cog.embedded_bundle, enforcing: false, calling_convention: :all
+  alias Cog.Commands.Helpers, as: CommandHelpers
+
+  @moduledoc """
+  Returns the specified fields from a set of inputs.
+  Be sure to add quotes when traversing a JSON path.
+
+  ## Example
+
+      @bot #{Cog.embedded_bundle}:rules --list --for-command="#{Cog.embedded_bundle}:permissions" | #{Cog.embedded_bundle}:pluck --fields=id
+      > { "id": "91edb472-04cf-4bca-ba05-e51b63f26758" }
+        { "id": "738ce922-14cd-4abc-fe94-81658707229c" }
+      @bot #{Cog.embedded_bundle}:rules --list --for-command="#{Cog.embedded_bundle}:permissions" | #{Cog.embedded_bundle}:pluck --fields=rule,command
+      > { "rule": "operable:manage_users",
+          "command": "operable:permissions" }
+        { "rule": "operable:manage_users",
+          "command": "operable:permissions" }
+      @bot #{Cog.embedded_bundle}:seed '[{"foo":{"bar.qux":{"baz":"stuff"}}}, {"foo": {"bar":{"baz":"me"}}}]' | #{Cog.embedded_bundle}:pluck --fields=foo"
+      > [ {"foo": {"bar.qux": {"baz": "stuff"} } }, {"foo": {"bar": {"baz": "me"} } } ]
+      @bot #{Cog.embedded_bundle}:seed '[{"foo":{"bar":{"baz":"stuff"}}}, {"foo": {"bar":{"baz":"me"}}}]' | #{Cog.embedded_bundle}:pluck --fields="foo.bar"
+      > { "foo.bar": {"baz": "stuff"} }
+        { "foo.bar": {"baz": "me"} }
+      @bot #{Cog.embedded_bundle}:seed '[{"foo":{"bar":{"baz":"stuff"}}, "qux": "one"}, {"foo": {"bar":{"baz":"me"}}, "qux": "two"}]' | #{Cog.embedded_bundle}:pluck --fields="qux,\"foo.bar\""
+      > { "qux": "one", "foo.bar": {"baz": "stuff"} }
+        { "qux": "two", "foo.bar": {"baz": "me"} }
+  """
+
+  option "fields", type: "string", required: false
+
+  defstruct req: nil, fields: nil, input: nil, output: nil, errors: []
+
+  def handle_message(req, state) do
+    case req |> validate |> execute |> format do
+      {:ok, data} ->
+        {:reply, req.reply_to, data, state}
+      {:error, error} ->
+        {:error, req.reply_to, error, state}
+    end
+  end
+
+  defp validate(%{cog_env: item}=req) do
+    %__MODULE__{req: req, input: item}
+    |> validate_options
+  end
+
+  defp validate_options(%__MODULE__{req: %{options: %{"fields" => opt_fields}}}=state) when not is_binary(opt_fields),
+    do: CommandHelpers.add_errors(state, {:not_string, opt_fields})
```

Comment:
Since the `fields` option is currently defined a as a string, is there a way that this command could receive a non-string value? Seems like something we might want to enforce before it gets to the command.